forc-pkg/src/manifest/manager.rs

```diff
@@ -0,0 +1,484 @@
+    // === Package Selection ===
+    pub package: Option<String>,
+    // === Source (Add only) ===
+    pub source_path: Option<String>,
+    pub git: Option<String>,
+    pub branch: Option<String>,
+    pub tag: Option<String>,
+    pub rev: Option<String>,
+    pub ipfs: Option<String>,
+    // === Section ===
+    pub contract_deps: bool,
+    pub salt: Option<String>,
+    // === IPFS Node ===
+    pub ipfs_node: Option<IPFSNode>,
+    // === Dependencies & Flags ===
+    pub dependencies: Vec<String>,
+    pub dry_run: bool,
+    pub offline: bool,
+    pub action: Action,
+}
+
+pub fn modify_dependencies(opts: ModifyOpts) -> Result<()> {
+    let cwd = if let Some(p) = opts.manifest_path.clone() {
+        PathBuf::from(p)
+    } else {
+        std::env::current_dir()?
+    };
+
+    let manifest_file = ManifestFile::from_dir(&cwd)?;
+    let root_dir = manifest_file.root_dir();
+    let mut member_manifests = manifest_file.member_manifests()?;
+
+    let package_spec_dir = resolve_package_path(
+        &manifest_file,
+        opts.package.clone(),
+        &root_dir,
+        &member_manifests,
+    )?;
+
+    let content = std::fs::read_to_string(&package_spec_dir)?;
+    let mut toml_doc = content.parse::<DocumentMut>()?;
+    let backup_doc = toml_doc.clone();
+
+    let mut package_spec = PackageManifestFile::from_file(&package_spec_dir)?;
+
+    let lock_path = package_spec.lock_path()?;
+    let old_lock = Lock::from_path(&lock_path).ok().unwrap_or_default();
+
+    let mut deps_regular = package_spec
+        .manifest
+        .dependencies
+        .take()
+        .unwrap_or_default();
+    let mut deps_contract = package_spec
+        .manifest
+        .contract_dependencies
+        .take()
+        .unwrap_or_default();
+
+    let mut section = if opts.contract_deps {
+        DepSection::Contract(&mut deps_contract, opts.salt.clone())
+    } else {
+        DepSection::Regular(&mut deps_regular)
+    };
+
+    match opts.action {
+        Action::Add => {
+            for dependency in &opts.dependencies {
+                let (dep_name, dependency_data) =
+                    resolve_dependency(dependency, &opts, &member_manifests, package_spec.dir())?;
+                section.insert_dep(dep_name, dependency_data, opts.salt.clone());
+            }
+
+            section.add_deps_manifest_table(
+                &mut toml_doc,
+                package_spec_dir.clone(),
+                &mut package_spec.manifest,
+            )?;
+        }
+        Action::Remove => {
+            let dep_refs: Vec<&str> = opts.dependencies.iter().map(String::as_str).collect();
+
+            section.remove_deps_manifest_table(
+                &mut toml_doc,
+                package_spec_dir.clone(),
+                &mut package_spec.manifest,
+                &dep_refs,
+            )?;
+        }
+    }
+
+    member_manifests.insert(
+        package_spec.project_name().to_string(),
+        package_spec.clone(),
```

Comment:
cant see var usage after this point; could remove clone?

```suggestion
        package_spec,
```

Reply:
Yes this works.